File: python/ema.py
class EMA:
    def __init__(self, b=0.9):
        self.b = b
        self.n = 0
        self.avg = 0

    def next(self, v):
        self.n += 1
        self.avg = self.b * v + (1 - self.b) * self.avg
        return self.avg

    def avg(self):
        return self.avg

File: rust/src/ema.rs
#[derive(Debug, Clone)]
pub struct EMA {
    b: f64,
    n: f64,
    avg: f64,
}
impl Default for EMA {
    fn default() -> Self {
        Self {
            b: 0.9,
            n: 0.,
            avg: 0.,
        }
    }
}
impl EMA {
    pub fn next(&mut self, v: f64) -> f64 {
        self.n += 1.;
        self.avg = self.b * v + (1. - self.b) * self.avg;
        self.avg
    }
    pub fn avg(&self) -> f64 {
        self.avg
    }
}